provider/aws_test.go

```diff
@@ -781,6 +781,12 @@ func TestAWSSuitableZones(t *testing.T) {
 	}{
 		{"foo.bar.example.org.", []*route53.HostedZone{zones["example-org-private"], zones["bar-example-org-private"], zones["bar-example-org"]}},
 		{"foo.example.org.", []*route53.HostedZone{zones["example-org-private"], zones["example-org"]}},
+
+		// bar.example.org is NOT suitable
+		{"foobar.example.org.", []*route53.HostedZone{zones["example-org-private"], zones["example-org"]}},
+
+		// all matching private zones are suitable (i'm not sure why)
```

Comment:
It's explained in https://github.com/kubernetes-incubator/external-dns/pull/356#issue-145757274 and your test matches the one above, so I think it's fine how you did it. Can you move this test line to first block so it's easier to see?

Reply:
Submitted in new PR since this was merged.  Feel free to close if no longer needed. https://github.com/kubernetes-incubator/external-dns/pull/608